pystac/extensions/version.py

```diff
@@ -0,0 +1,172 @@
+"""Implement the version extension.
+
+https://github.com/radiantearth/stac-spec/tree/dev/extensions/version
+"""
+
+import pystac
+from pystac import collection
+from pystac import Extensions
+from pystac import item
+from pystac import link
+from pystac.extensions import base
+
+VERSION = 'version'
+DEPRECATED = 'deprecated'
+LATEST = 'latest-version'
+PREDECESSOR = 'predecessor-version'
+SUCCESSOR = 'successor-version'
+
+# Media type for links.
+MEDIA_TYPE = 'application/json'
+
+
+class VersionItemExt(base.ItemExtension):
+    """Add an asset version string to a STAC Item."""
+    def __init__(self, an_item):
+        self.item = an_item
+
+    def apply(self, version, deprecated=None, latest=None, predecessor=None, successor=None):
+        self.version = version
+        if deprecated is not None:
+            self.deprecated = deprecated
+        if latest:
+            self.latest = latest
+        if predecessor:
+            self.predecessor = predecessor
+        if successor:
+            self.successor = successor
+
+    @property
+    def version(self):
+        return self.item.properties.get(VERSION)
+
+    @version.setter
+    def version(self, v):
+        self.item.properties[VERSION] = v
+
+    @property
+    def deprecated(self):
+        return bool(self.item.properties.get(DEPRECATED))
+
+    @deprecated.setter
+    def deprecated(self, v):
+        if not isinstance(v, bool):
+            raise pystac.STACError(DEPRECATED + ' must be a bool')
+        self.item.properties[DEPRECATED] = v
+
+    @property
+    def latest(self):
+        items = self.item.get_stac_objects(LATEST)
+        if items:
+            return next(items)
+
+    @latest.setter
+    def latest(self, source_item):
+        self.item.add_link(link.Link(LATEST, source_item, MEDIA_TYPE))
+
+    @property
+    def predecessor(self):
+        items = self.item.get_stac_objects(PREDECESSOR)
+        if items:
+            return next(items)
+
+    @predecessor.setter
+    def predecessor(self, source_item):
+        self.item.add_link(link.Link(PREDECESSOR, source_item, MEDIA_TYPE))
+
+    @property
+    def successor(self):
+        items = self.item.get_stac_objects(SUCCESSOR)
+        if items:
+            return next(items)
```

Comment:
One last little nitpick -

Making the return in the None case would be more explicit here and elsewhere, rather than not returning and relying on Python's default mechanism to serve None back to the caller.

Here and in the other iterator cases you could combine these lines into:

```suggestion
       return next(items, None)
```

I think it'd still be readable if you even one-lined it:

```
return next(self.item.get_stac_objects(SUCCESSOR), None)
```

Reply:
Awesome!  I haven't looked at next() in ages.  The default `None` is so much simpler.